docs/howtos/extending/magicgui.md

```diff
@@ -783,3 +787,36 @@ viewer.window.add_dock_widget(my_widg)
 
 As above to turn this into a [plugin widget contribution](widgets-contribution-guide),
 simply provide the class definition and add to the plugin manifest.
+
+#### `magicgui.widgets.create_widget`
+
+You might want to add a layer selection as [shown above](#parameter-annotations) into your highly customizable `QtWidgets.QWidget` that is always synchronized with the available {attr}`napari.types.ImageData` layers in your viewer. For this you can use the {func}`create_widget <magicgui.widgets.create_widget>` function as described in the following example.
+
+To synchronize the information between the napari viewer (i.e. the available layers) the function `reset_choices` of the dropdown widget needs to be manually called whenever the main widget is shown. In the following example, this is done by defining two class methods `showEvent` and `reset_choices`.
```

Comment:
I would have thought it needs to be called when the input selector widget is shown? If the main widget is open. then layers are changed, you'd expect the input selector widget to update?

Reply:
I tested it. Napari somehow internally calls `reset_choices` of the main widget. So `showEvent` and `reset_choices` need to be named exactly like that. I updated it in the PR